tests/test_object_detection.py

```diff
@@ -0,0 +1,448 @@
+    ):
+        assert (score == no_auxiliary_inputs_score).all()
+
+    badloc_issues_per_box = _find_label_issues_per_box(badloc_scores_per_box, BADLOC_THRESHOLD)
+    badloc_issues_per_image = _pool_box_scores_per_image(badloc_issues_per_box)
+    badloc_issues = np.sum(badloc_issues_per_image)
+    assert np.sum(badloc_issues_per_image[5:]) == 5  # check bad labels were detected correctly
+    assert badloc_issues == 5
+
+    swap_scores_per_box = _compute_swap_box_scores(
+        alpha=ALPHA,
+        high_probability_threshold=HIGH_PROBABILITY_THRESHOLD,
+        auxiliary_inputs=auxiliary_inputs,
+    )
+
+    swap_scores_no_auxillary_inputs = _compute_swap_box_scores(
+        alpha=ALPHA,
+        high_probability_threshold=HIGH_PROBABILITY_THRESHOLD,
+        labels=labels,
+        predictions=predictions,
+    )
+
+    for score, no_auxiliary_inputs_score in zip(
+        swap_scores_per_box, swap_scores_no_auxillary_inputs
+    ):
+        assert (score == no_auxiliary_inputs_score).all()
+
+    swap_issues_per_box = _find_label_issues_per_box(swap_scores_per_box, SWAP_THRESHOLD)
+    swap_issues_per_image = _pool_box_scores_per_image(swap_issues_per_box)
+    swap_issues = np.sum(swap_issues_per_image)
+    assert np.sum(swap_scores_per_box[2]) > np.sum(swap_scores_per_box[7])
+    assert swap_issues == 0
+
+    label_issues = find_label_issues(labels, predictions)
+    assert np.sum(label_issues) == np.sum(
+        (swap_issues_per_image + badloc_issues_per_image + overlooked_issues_per_image) > 0
+    )
+    assert np.sum(label_issues[5:]) == 5  # check bad labels were detected correctly
+
+    swap_issues_per_box = _find_label_issues_per_box(swap_scores_per_box, 0.7)
+    swap_issues_per_image = _pool_box_scores_per_image(swap_issues_per_box)
+    swap_issues = np.sum(swap_issues_per_image)
+    assert swap_issues == 4
+    assert np.sum(swap_issues_per_image[5:]) == 4  # check bad labels were detected correctly
+
+
+def test_separate_prediction():
+    pred_bboxes = np.array(
+        [
+            np.array(list(generate_bbox(300)) + [0.97]),
+            np.empty(shape=[0, 5], dtype=np.float32),
+            np.array(list(generate_bbox(300)) + [0.94]),
+        ],
+        dtype=object,
+    )
+    pred_labels = np.array([0, 2])
+    pred_probs = np.array([[0.98, 0.01, 0.01], [0.02, 0.02, 0.98]])
+    all_pred_prediction = np.array([pred_bboxes, pred_labels, pred_probs], dtype=object)
+    prediction_type = _get_prediction_type(all_pred_prediction)
+    assert prediction_type == "all_pred"
+
+    boxes, labels, pred_probs = _separate_prediction(
+        all_pred_prediction, prediction_type=prediction_type
+    )
+    assert len(labels) == len(pred_probs)
+
+
+def test_return_issues_ranked_by_scores():
+    label_issue_idx = find_label_issues(labels, predictions, return_indices_ranked_by_score=True)
+    assert [5, 6, 7, 8, 9] in label_issue_idx[:5]  # lower scores for bad examples
```

Comment:
Should we checking the order of these as well? Or are they all expected to be roughly tied in terms of label quality score?

Reply:
Examples should be random values as the bad examples are generated randomly.